techniques/system/distributePolicy/1.0/propagatePromises.st

```diff
@@ -160,44 +174,79 @@ bundle agent propagatePromises
         comment => "Copying list of nodes under this policy server",
         classes => if_else("nodeslist_copied", "could_not_copy_nodeslist");
 
+
+  commands:
+    policy_server.!root_server.rsync_promises::
+      "/usr/bin/rsync --archive --checksum --compress --sparse --delete ${server_info.cfserved}:${server_data}/ ${client_data}"
+        classes => if_else("promises_propagated", "could_not_propagate_promises");
+    policy_server.!root_server.rsync_sharedfiles::
+      "/usr/bin/rsync --archive --checksum --compress --sparse --delete ${server_info.cfserved}:&SHARED_FILES_FOLDER&/ &SHARED_FILES_FOLDER&/"
+        classes => if_else("files_propagated", "could_not_propagate_files");
+
   methods:
     root_server::
       "any" usebundle => rudder_common_reports_generic("DistributePolicy", "configure_ncf_config", "&TRACKINGKEY&", "Configure ncf", "None", "Configure configuration library");
 
     !root_server::
       "any" usebundle => rudder_common_report("DistributePolicy", "result_na", "&TRACKINGKEY&", "Configure ncf", "None", "Configuration of configuration library is not necessary on non root servers");
 
-    # Success if files are updated or not changed (kept or repaired).
-    # root server have only tools to be updated and others have tools,
-    # promises, masterfiles folder to be updated.
-    pass3.(((root_server.propagate_tools_ok)|(!root_server.propagate_tools_ok.(promises_propagated|empty_promises_to_propagate).(shared_files_propagated|sharedfiles_not_existent).masterfiles_propagated.ncf_local_promises_propagated.ncf_common_promises_propagated.nodeslist_copied)).!(propagate_tools_error|promises_to_propagate_not_copied|sharedfiles_not_copied|could_not_propagate_masterfiles|could_not_propagate_ncf_local_promise|could_not_propagate_ncf_common_promise|could_not_copy_nodeslist))::
-      "any" usebundle => rudder_common_report("DistributePolicy", "result_success", "&TRACKINGKEY&", "Propagate promises", "None", "All files have been propagated");
-    pass3.promises_to_propagate_not_copied::
-      "any" usebundle => rudder_common_report("DistributePolicy", "result_error", "&TRACKINGKEY&", "Propagate promises", "None", "Cannot propagate policy");
+    # Success if resources are updated or not changed (kept or repaired).
+    # root server have only tools and nodelist to be updated
+    # relays have tools, ncf, masterfiles nodelist and shared_files folder to be updated.
+    pass3.root_server.propagate_tools_ok.nodeslist_copied.!(propagate_tools_error|could_not_copy_nodeslist)::
+      "any" usebundle => rudder_common_report("DistributePolicy", "result_success", "&TRACKINGKEY&", "Synchronize resources", "None", "All resources have been updated");
```

Comment:
why is the message different?

Reply:
Because on root_server we don't synchronize them (they are local), but we rather move them around or define hash of their values